### PR TITLE
fix(effects): resubscribe every time and error occurs (#2023)

### LINKED ISSUE #2023: Resubscribe on Error should work multiple times
## Minimal reproduction of the bug/regression with instructions:

```
$myEffect = createEffect(() =>
  this.actions$.pipe(
    ofType('MY_TYPE'),
    tap(() => console.log('I am here')),
    switchMap(() => throwError('error'))
  )
)
```

The console log shows only twice.

On the first error it switches to a new observable, but without any safeguard for the next error.
https://github.com/ngrx/platform/blob/master/modules/effects/src/effects_resolver.ts#L32

## Expected behavior:

The resubscribe should work multiple times in case error happens in the stream
The error handling should be in a recursive function, something like this:

```
const handleErrors = obs$ => obs$.pipe(catchError(error => {
  if (errorHandler) errorHandler.handleError(error);
  // Return observable that produces this particular effect
  return handleErrors(obs$);
}));

const resubscribable$ = resubscribeOnError ? handleErrors(observable$) : observable$;
```

## Versions of NgRx, Angular, Node, affected browser(s) and operating system(s):

8.1.0

## I would be willing to submit a PR to fix this issue

[X] Yes (Assistance is provided if you need help submitting a pull request)
[ ] No

### PR DESCRIPTION
**Pull request by peterreisz**
## PR Type

What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Closes #2023

## What is the new behavior?

The effects will resubscribe every time an error occurs in the stream.

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: modules/effects/src/effects_resolver.ts [hunk 1]
```diff
@@ -24,14 +24,19 @@ export function mergeEffects(
           ? sourceInstance[propertyName]()
           : sourceInstance[propertyName];
 
+      const resubscribeInCaseError = (
```
Comment (timdeschryver) [inline modules/effects/src/effects_resolver.ts]:
```suggestion
      const resubscribeInOfCaseError = (
```

Comment (brandonroberts) [reply modules/effects/src/effects_resolver.ts]: `resubscribeInCaseOfError`?

Comment (peterreisz) [reply modules/effects/src/effects_resolver.ts]: I prefer the last one, fixed.

**Comment by peterreisz** [comment]
@brandonroberts @timdeschryver This PR was reverted due to some CI issue. Are you planning on fixing the issue? Can I help somehow?

**Comment by timdeschryver** [comment]
@peterreisz see #2165

**Pull request merged**